### PR TITLE
New: TypeScript config parser.

### PR DESCRIPTION
**Pull request by sarvaje**
New: `typescript-config-is-valid` rule.

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

File: src/lib/types/typescriptconfig.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+export type TypeScriptPaths = {
+    [key: string]: Array<string>;
+};
+
+export type TypeScriptPlugin = {
+    name: string;
+};
+
+export type TypeScriptCompilerOptions = {
+    allowJs: boolean;
```
Comment (alrra) [inline src/lib/types/typescriptconfig.ts]: @sarvaje Can we somehow get these automatically?

Comment (alrra) [reply src/lib/types/typescriptconfig.ts]: I'm referring to everything.

File: docs/user-guide/rules/typescript-config-is-valid.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# typescript-config-is-valid (`typescript-config-is-valid`)
+
+`typescript-config-is-valid` warns again providing an
+invalid typescript configuration file `tsconfig.json`.
```
Comment (alrra) [inline docs/user-guide/rules/typescript-config-is-valid.md]: > invalid TypeScript configuration file (i.e. `tsconfig.json`).

File: docs/user-guide/rules/typescript-config-is-valid.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## Why is this important?
+
+If you are bilding an app or a website with typescript, you
```
Comment (alrra) [inline docs/user-guide/rules/typescript-config-is-valid.md]: > using TypeScript

File: src/lib/types/typescriptconfig.ts [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    es6 = 'es6',
+    es2015 = 'es2015',
+    es2016 = 'es2016',
+    es2017 = 'es2017',
+    esnext = 'esnext',
+    none = 'none'
+}
+
+/**
+ * Moduel resolution values.
```
Comment (alrra) [inline src/lib/types/typescriptconfig.ts]: > Module

File: src/lib/types/typescriptconfig.ts (hunk 1 above)
Comment (sarvaje) [reply src/lib/types/typescriptconfig.ts]: As we talk the other day, it would be great if we get a packages with the types.

Comment (alrra) [reply src/lib/types/typescriptconfig.ts]: @sarvaje Sure, I will look into it.

**Comment by sarvaje** [comment]
Clossing in favor of https://github.com/sonarwhal/sonarwhal/pull/847

**Pull request closed without merging**